packages/addon/public/token-bridge.js

```diff
@@ -0,0 +1,79 @@
+const ALLOWED_ORIGINS = new Set([
+  "https://auth-stage.tb.pro",
+  "http://localhost:5173",           // dev
+  "http://127.0.0.1:5173"            // dev
+]);
+
+window.postMessage({ type: "TB/BRIDGE_READY" }, window.location.origin);
+console.log(`[🌉 token-bridge] the token bridge has loaded.`);
+
+// Visual cue, make sure to remove.
+const tag = document.createElement("div");
+tag.textContent = "✅ Content script injected";
+Object.assign(tag.style, {
+  position: "fixed", zIndex: 999999, inset: "8px auto auto 8px",
+  padding: "6px 10px", background: "lime", color: "black",
+  fontFamily: "monospace", boxShadow: "0 2px 8px rgba(0,0,0,.25)"
+});
+document.documentElement.appendChild(tag);
+
+// Initial message to the background
+browser.runtime.sendMessage({
+  type: "TB/PING",
+  text: "This got sent from the bridge to the background."
+});
+
+
+window.addEventListener("message", (e) => {
+  // if (e.origin !== APP_ORIGIN) return;   // security: only trust your app
+  // if (e.source !== window) return;       // same-page messages only
+  // if (!e.data || e.data.type !== "TB_PING") return;
```

Comment:
Security checks should be uncommented and updated with correct checks for message types and valid origins.